doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,952 @@
++     |        +--ro state
++     |           +--ro address?                 oc-inet:host
+      |           +--ro port?                    oc-inet:port-number
+      |           +--ro version?                 uint8
+      |           +--ro association-type?        enumeration
+      |           +--ro iburst?                  boolean
+      |           +--ro prefer?                  boolean
++     |           +--ro stratum?                 uint8
+      |           +--ro root-delay?              uint32
+      |           +--ro root-dispersion?         uint64
+      |           +--ro offset?                  uint64
++     |           +--ro poll-interval?           uint32
++     |           +--rw oc-sys-ext:key-id?       uint16
++     |           +--ro oc-sys-ext:peerdelay?    decimal64
++     |           +--ro oc-sys-ext:peeroffset?   decimal64
++     |           +--ro oc-sys-ext:peerjitter?   decimal64
++     |           +--ro oc-sys-ext:selmode?      string
++     |           +--ro oc-sys-ext:refid?        inet:host
++     |           +--ro oc-sys-ext:peertype?     string
++     |           +--ro oc-sys-ext:now?          uint32
++     |           +--ro oc-sys-ext:reach?        uint8
+```
+      
+```diff
+module: sonic-system-ntp
++     +--rw NTP
++     |  +--rw NTP_LIST* [global_key]
++     |     +--rw global_key      enumeration
++     |     +--rw src_intf*       union
++     |     +--rw vrf?            union
++     |     +--rw auth_enabled?   boolean
++     |     +--rw trustedkeys*    -> /sonic-system-ntp/NTP_AUTHENTICATION_KEY/NTP_AUTHENTICATION_KEY_LIST/key_id
++     +--rw NTP_AUTHENTICATION_KEY
++     |  +--rw NTP_AUTHENTICATION_KEY_LIST* [key_id]
++     |     +--rw key_id           uint16
++     |     +--rw key_type         enumeration
++     |     +--rw key_value        string
++     |     +--rw key_encrypted?   boolean
++     +--rw NTP_SERVER
++        +--rw NTP_SERVER_LIST* [server_address]
++           +--rw server_address    inet:host
++           +--rw key_id?           -> /sonic-system-ntp/NTP_AUTHENTICATION_KEY/NTP_AUTHENTICATION_KEY_LIST/key_id
+```
+
+### 3.6.2 CLI
+
+
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+
+sonic(config)# ntp
+  authenticate        Authenticate time sources
+  authentication-key  Authentication key for trusted time sources
+  server              Configure NTP server
+  source-interface    Configure NTP source interface to pick the source IP, used for the NTP packets
+  trusted-key         Key numbers for trusted time sources
+  vrf                 Enable NTP on VRF
+
+```
+
+##### 3.6.2.1.1 Configure NTP server
+```
+sonic(config)#ntp
+    server              Configure NTP server
+sonic(config)#ntp server 
+String  NTP server address or name
+
+sonic(config)# ntp server 10.11.0.1
+
+sonic(config)# ntp server 2001:aa:aa::a
+
+sonic(config)# ntp server pool.ntp.org
+
+```

```

Comment:
Getting comments: "HI Arun
how many max ntp servers we can configure ?
i cannot find this in below hld
" 
I found this link on ntp.conf advising on the number of NTP servers. (http://support.ntp.org/bin/view/Support/SelectingOffsiteNTPServers)  Section  5.3.4. Excessive Number of Upstream Time Servers